src/main/scala/de/frosner/ddq/core/Check.scala

```diff
@@ -563,6 +573,41 @@ object Check {
   )
 
   /**
+   * Check whether columns in the dependent set have functional dependency on determinant set.
+   * @param determinantSet sequence of column names which form a determinant set
+   * @param dependentSet sequence of column names which form a dependent set
+   * @return [[core.Constraint]] object
+   */
+  def hasFunctionalDependency(determinantSet: Seq[String], dependentSet: Seq[String]) = Constraint(
+    df => {
+      val determinantIndexes = df.columns.zipWithIndex.filter {
+        case (name, index) => determinantSet.contains(name)
+      }.map { case (name, index) => index }
+      val dependentIndexes = df.columns.zipWithIndex.filter {
+        case (name, index) => dependentSet.contains(name)
+      }.map { case (name, index) => index }
+      val rdd = df.map(row => (
+        determinantIndexes.map(index => row(index)).toVector,
+        dependentIndexes.map(index => row(index)).toVector)).
+        aggregateByKey(Set(): Set[Any])((acc, el) => acc + el, (s1, s2) => s1 | s2)
```

Comment:
Consider using Set.empty instead of Set(). It will directly give you an empty set instance that is shared across your program in a type safe way instead of going through a lot of transformations and returning you a new instance every time.